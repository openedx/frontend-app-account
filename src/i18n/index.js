// Placeholder be overridden by `make pull_translations`
export default {
  ar: {},
  'zh-hk': {},
  'zh-cn': {},
  uk: {},
  'tr-tr': {},
  th: {},
  te: {},
  ru: {},
  'pt-pt': {},
  'pt-br': {},
  'it-it': {},
  id: {},
  hi: {},
  he: {},
  'fr-ca': {},
  fa: {},
  'es-es': {},
  'es-419': {},
  el: {},
  'de-de': {},
  da: {},
  bo: {},
};
