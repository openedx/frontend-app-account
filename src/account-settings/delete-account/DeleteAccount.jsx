import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { getConfig } from '@edx/frontend-platform';
import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
import { Button, Hyperlink } from '@openedx/paragon';

// Actions
import {
  deleteAccount,
  deleteAccountConfirmation,
  deleteAccountFailure,
  deleteAccountReset,
  deleteAccountCancel,
} from './data/actions';

// Messages
import messages from './messages';

// Components
import ConnectedConfirmationModal from './ConfirmationModal';
import PrintingInstructions from './PrintingInstructions';
import ConnectedSuccessModal from './SuccessModal';
import BeforeProceedingBanner from './BeforeProceedingBanner';

export class DeleteAccount extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      password: '',
    };
  }

  handleSubmit = () => {
    if (this.state.password === '') {
      this.props.deleteAccountFailure('empty-password');
    } else {
      this.props.deleteAccount(this.state.password);
    }
  };

  handleCancel = () => {
    this.setState({ password: '' });
    this.props.deleteAccountCancel();
  };

  handlePasswordChange = (e) => {
    this.setState({ password: e.target.value.trim() });
    this.props.deleteAccountReset();
  };

  handleFinalClose = () => {
    global.location = getConfig().LOGOUT_URL;
  };

  render() {
    const {
      hasLinkedTPA, isVerifiedAccount, status, errorType, intl,
    } = this.props;
    const canDelete = isVerifiedAccount && !hasLinkedTPA;
    const supportArticleUrl = process.env.SUPPORT_URL_TO_UNLINK_SOCIAL_MEDIA_ACCOUNT;

    // TODO: We lack a good way of providing custom language for a particular site.  This is a hack
    // to allow edx.org to fulfill its business requirements.
    const deleteAccountText2MessageKey = getConfig().SITE_NAME === 'edX'
      ? 'account.settings.delete.account.text.2.edX'
      : 'account.settings.delete.account.text.2';

    const optInInstructionMessageId = getConfig().MARKETING_EMAILS_OPT_IN
      ? 'account.settings.delete.account.please.confirm'
      : 'account.settings.delete.account.please.activate';

    return (
      <div>
        <h2 className="section-heading h4 mb-3">
          {intl.formatMessage(messages['account.settings.delete.account.header'])}
        </h2>
        {
          this.props.canDeleteAccount ? (
            <>
              <p>{intl.formatMessage(messages['account.settings.delete.account.subheader'])}</p>
              <p>
                {intl.formatMessage(
                  messages['account.settings.delete.account.text.1'],
                  { siteName: getConfig().SITE_NAME },
                )}
              </p>
              <p>
                {intl.formatMessage(
                  messages[deleteAccountText2MessageKey],
                  { siteName: getConfig().SITE_NAME },
                )}
              </p>
              <p>
                <PrintingInstructions />
              </p>
              <p className="text-danger h6">
                {intl.formatMessage(
                  messages['account.settings.delete.account.text.warning'],
                  { siteName: getConfig().SITE_NAME },
                )}
              </p>
              <p>
                <Hyperlink destination="https://help.edx.org/edxlearner/s/topic/0TOQq0000001UdZOAU/account-basics">
                  {intl.formatMessage(messages['account.settings.delete.account.text.change.instead'])}
                </Hyperlink>
              </p>
              <p>
                <Button
                  variant="outline-danger"
                  onClick={canDelete ? this.props.deleteAccountConfirmation : null}
                  disabled={!canDelete}
                >
                  {intl.formatMessage(messages['account.settings.delete.account.button'])}
                </Button>
              </p>
              {isVerifiedAccount ? null : (
                <BeforeProceedingBanner
                  instructionMessageId={optInInstructionMessageId}
                  supportArticleUrl="https://support.edx.org/hc/en-us/articles/115000940568-How-do-I-confirm-my-email"
                />
              )}
              {hasLinkedTPA ? (
                <BeforeProceedingBanner
                  instructionMessageId="account.settings.delete.account.please.unlink"
                  supportArticleUrl={supportArticleUrl}
                />
              ) : null}

              <ConnectedConfirmationModal
                status={status}
                errorType={errorType}
                onSubmit={this.handleSubmit}
                onCancel={this.handleCancel}
                onChange={this.handlePasswordChange}
                password={this.state.password}
              />

              <ConnectedSuccessModal status={status} onClose={this.handleFinalClose} />
            </>
          ) : (
            <p>{intl.formatMessage(messages['account.settings.cannot.delete.account.text'])}</p>
          )
        }

      </div>
    );
  }
}

DeleteAccount.propTypes = {
  deleteAccount: PropTypes.func.isRequired,
  deleteAccountConfirmation: PropTypes.func.isRequired,
  deleteAccountFailure: PropTypes.func.isRequired,
  deleteAccountReset: PropTypes.func.isRequired,
  deleteAccountCancel: PropTypes.func.isRequired,
  status: PropTypes.oneOf(['confirming', 'pending', 'deleted', 'failed']),
  errorType: PropTypes.oneOf(['empty-password', 'server']),
  hasLinkedTPA: PropTypes.bool,
  isVerifiedAccount: PropTypes.bool,
  canDeleteAccount: PropTypes.bool,
  intl: intlShape.isRequired,
};

DeleteAccount.defaultProps = {
  hasLinkedTPA: false,
  isVerifiedAccount: true,
  status: null,
  errorType: null,
  canDeleteAccount: true,
};

// Assume we're part of the accountSettings state.
const mapStateToProps = state => state.accountSettings.deleteAccount;

export default connect(
  mapStateToProps,
  {
    deleteAccount,
    deleteAccountConfirmation,
    deleteAccountFailure,
    deleteAccountReset,
    deleteAccountCancel,
  },
)(injectIntl(DeleteAccount));
