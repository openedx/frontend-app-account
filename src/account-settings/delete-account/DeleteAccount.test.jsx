/* eslint-disable react/jsx-no-useless-fragment */
import React from 'react';
import renderer from 'react-test-renderer';
import { IntlProvider, injectIntl } from '@edx/frontend-platform/i18n';

// Testing the modals separately, they just clutter up the snapshots if included here.
jest.mock('./ConfirmationModal', () => function ConfirmationModalMock() {
  return <></>;
});
jest.mock('./SuccessModal', () => function SuccessModalMock() {
  return <></>;
});

import { DeleteAccount } from './DeleteAccount'; // eslint-disable-line import/first

const IntlDeleteAccount = injectIntl(DeleteAccount);

describe('DeleteAccount', () => {
  let props = {};

  beforeEach(() => {
    props = {
      deleteAccount: jest.fn(),
      deleteAccountConfirmation: jest.fn(),
      deleteAccountFailure: jest.fn(),
      deleteAccountReset: jest.fn(),
      deleteAccountCancel: jest.fn(),
      status: null,
      errorType: null,
      hasLinkedTPA: false,
      isVerifiedAccount: true,
    };
  });

  it('should match default section snapshot', () => {
    const tree = renderer
      .create((
        <IntlProvider locale="en">
          <IntlDeleteAccount
            {...props}
          />
        </IntlProvider>
      ))
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should match unverified account section snapshot', () => {
    const tree = renderer
      .create((
        <IntlProvider locale="en">
          <IntlDeleteAccount
            {...props}
            isVerifiedAccount={false}
          />
        </IntlProvider>
      ))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should match unverified account section snapshot', () => {
    const tree = renderer
      .create((
        <IntlProvider locale="en">
          <IntlDeleteAccount
            {...props}
            hasLinkedTPA
          />
        </IntlProvider>
      ))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
