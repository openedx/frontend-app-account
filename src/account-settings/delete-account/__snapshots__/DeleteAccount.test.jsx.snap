// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteAccount should match default section snapshot 1`] = `
<div>
  <h2
    className="section-heading h4 mb-3"
  >
    Delete My Account
  </h2>
  <p>
    We're sorry to see you go!
  </p>
  <p>
    Please note: Deletion of your account and personal data is permanent and cannot be undone. localhost will not be able to recover your account or the data that is deleted.
  </p>
  <p>
    Once your account is deleted, you cannot use it to take courses on localhost.
  </p>
  <p>
    You may also lose access to verified certificates and other program credentials. You can make a copy of these for your records before proceeding with deletion.
  </p>
  <p
    className="text-danger h6"
  >
    Warning: Account deletion is permanent. Please read the above carefully before proceeding. This is an irreversible action, and you will no longer be able to use the same email on localhost.
  </p>
  <p>
    <a
      className="pgn__hyperlink default-link standalone-link"
      href="https://support.edx.org/hc/en-us/sections/115004139268-Manage-Your-Account-Settings"
      onClick={[Function]}
      target="_self"
    >
      Want to change your email, name, or password instead?
    </a>
  </p>
  <p>
    <button
      className="btn btn-outline-danger"
      disabled={false}
      onClick={[MockFunction]}
      type="button"
    >
      Delete My Account
    </button>
  </p>
</div>
`;

exports[`DeleteAccount should match unverified account section snapshot 1`] = `
<div>
  <h2
    className="section-heading h4 mb-3"
  >
    Delete My Account
  </h2>
  <p>
    We're sorry to see you go!
  </p>
  <p>
    Please note: Deletion of your account and personal data is permanent and cannot be undone. localhost will not be able to recover your account or the data that is deleted.
  </p>
  <p>
    Once your account is deleted, you cannot use it to take courses on localhost.
  </p>
  <p>
    You may also lose access to verified certificates and other program credentials. You can make a copy of these for your records before proceeding with deletion.
  </p>
  <p
    className="text-danger h6"
  >
    Warning: Account deletion is permanent. Please read the above carefully before proceeding. This is an irreversible action, and you will no longer be able to use the same email on localhost.
  </p>
  <p>
    <a
      className="pgn__hyperlink default-link standalone-link"
      href="https://support.edx.org/hc/en-us/sections/115004139268-Manage-Your-Account-Settings"
      onClick={[Function]}
      target="_self"
    >
      Want to change your email, name, or password instead?
    </a>
  </p>
  <p>
    <button
      className="btn btn-outline-danger"
      disabled={true}
      onClick={null}
      type="button"
    >
      Delete My Account
    </button>
  </p>
  <div
    className="alert d-flex align-items-start alert-warning mt-n2"
  >
    <div>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-triangle-exclamation mr-2"
        data-icon="triangle-exclamation"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={{}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 32c14.2 0 27.3 7.5 34.5 19.8l216 368c7.3 12.4 7.3 27.7 .2 40.1S486.3 480 472 480L40 480c-14.3 0-27.6-7.7-34.7-20.1s-7-27.8 .2-40.1l216-368C228.7 39.5 241.8 32 256 32zm0 128c-13.3 0-24 10.7-24 24l0 112c0 13.3 10.7 24 24 24s24-10.7 24-24l0-112c0-13.3-10.7-24-24-24zm32 224a32 32 0 1 0 -64 0 32 32 0 1 0 64 0z"
          fill="currentColor"
          style={{}}
        />
      </svg>
    </div>
    <div>
      Before proceeding, please 
      <a
        className="pgn__hyperlink default-link standalone-link"
        href="https://support.edx.org/hc/en-us/articles/115000940568-How-do-I-confirm-my-email-"
        onClick={[Function]}
        target="_self"
      >
        activate your account
      </a>
      .
    </div>
  </div>
</div>
`;

exports[`DeleteAccount should match unverified account section snapshot 2`] = `
<div>
  <h2
    className="section-heading h4 mb-3"
  >
    Delete My Account
  </h2>
  <p>
    We're sorry to see you go!
  </p>
  <p>
    Please note: Deletion of your account and personal data is permanent and cannot be undone. localhost will not be able to recover your account or the data that is deleted.
  </p>
  <p>
    Once your account is deleted, you cannot use it to take courses on localhost.
  </p>
  <p>
    You may also lose access to verified certificates and other program credentials. You can make a copy of these for your records before proceeding with deletion.
  </p>
  <p
    className="text-danger h6"
  >
    Warning: Account deletion is permanent. Please read the above carefully before proceeding. This is an irreversible action, and you will no longer be able to use the same email on localhost.
  </p>
  <p>
    <a
      className="pgn__hyperlink default-link standalone-link"
      href="https://support.edx.org/hc/en-us/sections/115004139268-Manage-Your-Account-Settings"
      onClick={[Function]}
      target="_self"
    >
      Want to change your email, name, or password instead?
    </a>
  </p>
  <p>
    <button
      className="btn btn-outline-danger"
      disabled={true}
      onClick={null}
      type="button"
    >
      Delete My Account
    </button>
  </p>
  <div
    className="alert d-flex align-items-start alert-warning mt-n2"
  >
    <div>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-triangle-exclamation mr-2"
        data-icon="triangle-exclamation"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={{}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 32c14.2 0 27.3 7.5 34.5 19.8l216 368c7.3 12.4 7.3 27.7 .2 40.1S486.3 480 472 480L40 480c-14.3 0-27.6-7.7-34.7-20.1s-7-27.8 .2-40.1l216-368C228.7 39.5 241.8 32 256 32zm0 128c-13.3 0-24 10.7-24 24l0 112c0 13.3 10.7 24 24 24s24-10.7 24-24l0-112c0-13.3-10.7-24-24-24zm32 224a32 32 0 1 0 -64 0 32 32 0 1 0 64 0z"
          fill="currentColor"
          style={{}}
        />
      </svg>
    </div>
    <div>
      Before proceeding, please 
      <a
        className="pgn__hyperlink default-link standalone-link"
        href="https://support.edx.org/hc/en-us/articles/207206067"
        onClick={[Function]}
        target="_self"
      >
        unlink all social media accounts
      </a>
      .
    </div>
  </div>
</div>
`;
